### PR TITLE
More debugger

### PR DESCRIPTION
**Pull request by Malabarba**
- Debugger's eval and inject also accept code non-interactively. Instead of sending :eval on the message's "input" entry, send something like `{:response :eval, :code 123}`. This allows cider.el to integrate regular eval commands with the debugger.  
  *In short*, commands like `C-x C-e` or `C-c M-:` will evaluate code in the lexical context while debug-mode is active.
- Since the order inside maps is unreliable, try to use map keys as a coordinate. Affects #211.  
  cider-debug-map
- There are now commands for inspecting values in the lexical context. `:inspect` and `:locals`, in particular.  
  cider-debug-inspect-locals
- When a form has no :cider-coor metadata, but it is a list and its `first` *does* have that metadata, this is probably a proper code form built by a macro (like the threading macros). The debugger will now use the meta of the `first` to guess the coordinates of the parent form.  
  In other words: added support for `->`, `->>`, etc.  
  cider-debug-threading-macro

### CODE REVIEW

**Comment by Malabarba** [comment]
Screenshots to come.

**Comment by Malabarba** [comment]
Screenshots are up, and the tests should be silenced now.

**Comment by bbatsov** [comment]
The build is still failing for some profiles, though.

**Comment by Malabarba** [comment]
Yeah, I'm looking into that.

File: src/cider/nrepl/middleware/debug.clj [hunk 1]
```diff
@@ -160,22 +171,31 @@
...
     out: Skip breakpoints in the current sexp.
+    inspect: Evaluate an expression and inspect it.
+    locals: Inspect local variables.
     inject: Evaluate an expression and return it.
     eval: Evaluate an expression, display result, and prompt again.
-    quit: Abort current eval session."
+    quit: Abort current eval session.
+
+  Response received can be any one of these values, or a map
+  whole :response entry is one of these values."
```
Comment (expez) [inline src/cider/nrepl/middleware/debug.clj]: This sentence is slightly mangled

Comment (Malabarba) [inline src/cider/nrepl/middleware/debug.clj]: Indeed. Fixed.

**Comment by Malabarba** [comment]
At last, I have defeated cloveral.

**Comment by bbatsov** [comment]
:+1:

File: test/clj/cider/nrepl/middleware/debug_test.clj [hunk 2]
```diff
@@ -103,3 +103,44 @@
...
+
+(deftest breakpoint-reader
+  (is (:cider-breakfunction (meta (d/breakpoint-reader '[a b c]))))
+  (is (= '[a :b 10 "ok"]
+         (remove #(:cider-breakfunction (meta %)) (d/breakpoint-reader '[a :b 10 "ok"]))))
+  ;; Just don't error
+  (is (map d/breakpoint-reader '[a :b 10 "ok"])))
+
+(deftest reader-macros
+  (binding [*data-readers* {'dbg d/debug-reader}]
```
Comment (expez) [inline test/clj/cider/nrepl/middleware/debug_test.clj]: @Malabarba If user code is evaluated in this context then the correct thing to do is to merge these two maps, not doing a replacement.
